src/udev/udev-node.c

```diff
@@ -193,43 +197,58 @@ static int link_find_prioritized(sd_device *dev, bool add, const char *stackdir,
 
 static size_t escape_path(const char *src, char *dest, size_t size) {
         size_t i, j;
+        uint64_t h;
 
         assert(src);
         assert(dest);
 
         for (i = 0, j = 0; src[i] != '\0'; i++) {
                 if (src[i] == '/') {
-                        if (j+4 >= size) {
-                                j = 0;
-                                break;
-                        }
+                        if (j+4 >= size)
+                                goto toolong;
                         memcpy(&dest[j], "\\x2f", 4);
                         j += 4;
                 } else if (src[i] == '\\') {
-                        if (j+4 >= size) {
-                                j = 0;
-                                break;
-                        }
+                        if (j+4 >= size)
+                                goto toolong;
                         memcpy(&dest[j], "\\x5c", 4);
                         j += 4;
                 } else {
-                        if (j+1 >= size) {
-                                j = 0;
-                                break;
-                        }
+                        if (j+1 >= size)
+                                goto toolong;
                         dest[j] = src[i];
                         j++;
                 }
         }
         dest[j] = '\0';
         return j;
+
+toolong:
+        h = siphash24_string(src, UDEV_NODE_HASH_KEY.bytes);
+
+        assert(strlen(dest));
+
+        dest[size - 12] = urlsafe_base64char((h >> 60) & 63);
+        dest[size - 11] = urlsafe_base64char((h >> 54) & 63);
+        dest[size - 10] = urlsafe_base64char((h >> 48) & 63);
+        dest[size -  9] = urlsafe_base64char((h >> 42) & 63);
+        dest[size -  8] = urlsafe_base64char((h >> 36) & 63);
+        dest[size -  7] = urlsafe_base64char((h >> 30) & 63);
+        dest[size -  6] = urlsafe_base64char((h >> 24) & 63);
+        dest[size -  5] = urlsafe_base64char((h >> 18) & 63);
+        dest[size -  4] = urlsafe_base64char((h >> 12) & 63);
+        dest[size -  3] = urlsafe_base64char((h >>  6) & 63);
+        dest[size -  2] = urlsafe_base64char( h        & 63);
```

Comment:
maybe intead:

```c
for (unsigned i = 0; i < 12; i++) 
        dest[size-i] = urlsafe_base64char((60 - i*6) & 63);
```

or so? (untested)

Reply:
maybe add a test case for this, that verifies this is stable and looks ok for even complex paths